pkg/kubelet/kuberuntime/image.go

```diff
@@ -0,0 +1,123 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kuberuntime
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/credentialprovider"
+	runtimeApi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/util/parsers"
+)
+
+// PullImage pulls an image from the network to local storage using the supplied
+// secrets if necessary.
+func (m *kubeGenericRuntimeManager) PullImage(image kubecontainer.ImageSpec, pullSecrets []api.Secret) error {
+	img := image.Image
+	repoToPull, _, _, err := parsers.ParseImageName(img)
+	if err != nil {
+		return err
+	}
+
+	keyring, err := credentialprovider.MakeDockerKeyring(pullSecrets, m.keyring)
+	if err != nil {
+		return err
+	}
+
+	creds, ok := keyring.Lookup(repoToPull)
+	if !ok {
+		glog.V(3).Infof("Pulling image %s without credentials", img)
+	}
+
+	imgSpec := &runtimeApi.ImageSpec{Image: &img}
+	var auth *runtimeApi.AuthConfig
+	if len(creds) > 0 {
+		authConfig := creds[0].AuthConfig
+		auth = &runtimeApi.AuthConfig{
+			Username:      &authConfig.Username,
+			Password:      &authConfig.Password,
+			Auth:          &authConfig.Auth,
+			ServerAddress: &authConfig.ServerAddress,
+			IdentityToken: &authConfig.IdentityToken,
+			RegistryToken: &authConfig.RegistryToken,
+		}
+	}
+
+	err = m.imageService.PullImage(imgSpec, auth)
+	if err != nil {
+		glog.Errorf("Pulling image %s failed: %v", img, err)
+		return err
+	}
+
+	return nil
+}
+
+// IsImagePresent checks whether the container image is already in the local storage.
+func (m *kubeGenericRuntimeManager) IsImagePresent(image kubecontainer.ImageSpec) (bool, error) {
+	images, err := m.imageService.ListImages(&runtimeApi.ImageFilter{
+		Image: &runtimeApi.ImageSpec{
+			Image: &image.Image,
+		},
+	})
+	if err != nil {
+		glog.Errorf("ListImages failed: %v", err)
+		return false, err
+	}
+
+	return len(images) > 0, nil
+}
+
+// ListImages gets all images currently on the machine.
+func (m *kubeGenericRuntimeManager) ListImages() ([]kubecontainer.Image, error) {
+	var images []kubecontainer.Image
+
+	allImages, err := m.imageService.ListImages(nil)
+	if err != nil {
+		glog.Errorf("ListImages failed: %v", err)
+		return nil, err
+	}
+
+	for _, img := range allImages {
+		images = append(images, kubecontainer.Image{
+			ID:          img.GetId(),
+			Size:        int64(img.GetSize_()),
+			RepoTags:    img.RepoTags,
+			RepoDigests: img.RepoDigests,
+		})
+	}
+
+	return images, nil
+}
+
+// Removes the specified image.
```

Comment:
nit: Add the full method name in the comment to be consistent with the methods above.